hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -246,11 +246,26 @@ public org.apache.iceberg.Table registerTable(TableIdentifier identifier, String
     TableOperations ops = newTableOps(identifier);
     InputFile metadataFile = fileIO.newInputFile(metadataFileLocation);
     TableMetadata metadata = TableMetadataParser.read(ops.io(), metadataFile);
+
+    if (!isMetastoreTable(metadataFileLocation)) {
+      // Metastore table and filesystem tables will have different naming conventions as per the spec.
```

Comment:
This seems unnecessary. Why not just update `parseVersion` to return `-1` if the version isn't valid? I think that's a better solution than rewriting the metadata.

Reply:
I like this suggestion. For filesystem tables, along with fixing register problem for hive, it can fix register problem for other catalogs also once they support it.